programs/flobrij/src/lib.rs

```diff
@@ -46,6 +46,67 @@ pub mod flobrij {
 
         Ok(())
     }
+
+    pub fn create_supportlevel(
+        ctx: Context<CreateSupportLevel>,
+        title: String,
+        monthly_price: u64,
+        description: String,
+        benefit: String,
+    ) -> ProgramResult {
+        let supportlevel: &mut Account<SupportLevel> = &mut ctx.accounts.supportlevel;
+        let payer: &Signer = &ctx.accounts.payer;
+
+        supportlevel.payer = *payer.key;
+
+        if title.chars().count() > 50 {
+            return Err(ErrorCode::TitleTooLong.into());
+        }
+
+        if description.chars().count() > 254 {
+            return Err(ErrorCode::DescriptionTooLong.into());
+        }
+
+        if benefit.chars().count() > 100 {
+            return Err(ErrorCode::BenefitTooLong.into());
+        }
+
+        supportlevel.title = title;
+        supportlevel.monthly_price = monthly_price;
+        supportlevel.description = description;
+        supportlevel.benefit = benefit;
+
+        Ok(())
+    }
+
+    pub fn set_supportlevel(
+        ctx: Context<SetSupportLevel>,
+        title: String,
+        monthly_price: u64,
```

Comment:
I think we should make this just "price" and then add another field to set how long a given  support level is active for. We can make it monthly in the front end, but for the backend seems like it should be generalized.

Reply:
Sounds good, I'll rename to `price`.

As far as "how long active", should we just run with an "active" boolean, and the Creator can just toggle that off when done? ("active" would either mean Creator "deleted/inactivated it" or it expired)